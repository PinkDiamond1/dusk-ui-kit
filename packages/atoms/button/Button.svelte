<script>
  import { setContext, getContext } from "svelte";
  import { variants, sizes, contexts } from "@dusk-network/helpers";
  import "./styles.css";

  export let value = false;
  export let variant = variants.BUTTON.BRAND;
  export let outline = false;
  export let size = sizes.BUTTON.DEFAULT;
  export let href = null;
  export let circle = false;
  export let disabled = false;
  export let active = false;

  const context = getContext("DUK:button:context");

  setContext("DUK:label:context", contexts.LABEL.BUTTON);
  setContext("DUK:icon:context", contexts.ICON.BUTTON);
</script>

{#if href && !disabled}
  <a href="{href}">
    <button
      class="{$$props.class || ''} duk-button"
      class:duk-button--brand="{variant === variants.BUTTON.BRAND}"
      class:duk-button--cta="{variant === variants.BUTTON.CTA}"
      class:duk-button--info="{variant === variants.BUTTON.INFO}"
      class:duk-button--success="{variant === variants.BUTTON.SUCCESS}"
      class:duk-button--warning="{variant === variants.BUTTON.WARNING}"
      class:duk-button--danger="{variant === variants.BUTTON.DANGER}"
      class:duk-button--small="{size === sizes.BUTTON.SMALL}"
      class:duk-button--large="{size === sizes.BUTTON.LARGE}"
      class:duk-button--outline="{outline}"
      class:duk-button--circle="{circle}"
      class:duk-button--active="{active}"
      class:duk-card-action-button="{context === contexts.BUTTON.ACTION.CARD}"
      class:duk-dialog-action-button="{context === contexts.BUTTON.ACTION.DIALOG}"
      class:duk-accordion__action="{context === contexts.BUTTON.ACCORDION}"
      disabled="{disabled || undefined}"
      on:click="{() => (value = !value)}"
      on:click
      on:mouseover
      on:*
    >
      <slot />
    </button>
  </a>
{:else}
  <button
    class="{$$props.class || ''} duk-button"
    class:duk-button--brand="{variant === variants.BUTTON.BRAND}"
    class:duk-button--cta="{variant === variants.BUTTON.CTA}"
    class:duk-button--info="{variant === variants.BUTTON.INFO}"
    class:duk-button--success="{variant === variants.BUTTON.SUCCESS}"
    class:duk-button--warning="{variant === variants.BUTTON.WARNING}"
    class:duk-button--danger="{variant === variants.BUTTON.DANGER}"
    class:duk-button--small="{size === sizes.BUTTON.SMALL}"
    class:duk-button--large="{size === sizes.BUTTON.LARGE}"
    class:duk-button--outline="{outline}"
    class:duk-button--circle="{circle}"
    class:duk-button--active="{active}"
    class:duk-card-action-button="{context === contexts.BUTTON.ACTION.CARD}"
    class:duk-dialog-action-button="{context === contexts.BUTTON.ACTION.DIALOG}"
    class:duk-accordion__action="{context === contexts.BUTTON.ACCORDION}"
    disabled="{disabled || undefined}"
    on:click="{() => (value = !value)}"
    on:click
    on:mouseover
    on:*
  >
    <slot />
  </button>
{/if}
