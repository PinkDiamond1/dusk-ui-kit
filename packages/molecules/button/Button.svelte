<script>
  // FIXME This needs DRYing (with a ButtonSkeleton.svelte file).
  import { setContext, getContext } from "svelte";
  import variants from "@dusk-network/helpers/variants.js";
  import sizes from "@dusk-network/helpers/sizes.js";
  import contexts from "@dusk-network/helpers/contexts.js";
  import "./styles.css";

  /**
   * Sets the `id` of the Button if set, otherwise the ID is generated.
   * @type {string}
   */
  export let id = "__DUK-button" + Math.random().toString(36);

  /**
   * Used as a switch when the button is clicked.
   */
  export let value = false;

  /**
   * Sets the variant of the Button.
   * @type {"brand" | "cta" | "info" | "success" | "warning" | "danger" }
   */
  export let variant = variants.BUTTON.BRAND;

  /**
   * Sets the size of the Button.
   * @type {"sm" | "lg" }
   */
  export let size = sizes.BUTTON.DEFAULT;

  /**
   * Sets the Button to outlined when true.
   */
  export let outline = false;

  /**
   * Sets the Button to have circular edges when true.
   */
  export let circle = false;

  /**
   * Sets the Button to act as a block element when true.
   */
  export let block = false;

  /**
   * Renders the Button as an `a` tag when set.
   */
  export let href = "";

  /**
   * Sets the `disabled` attribute on the Button when true.
   */
  export let disabled = false;

  const context = getContext("DUK:button:context");

  setContext("DUK:icon:context", contexts.ICON.BUTTON);
</script>

<div class="duk-button-wrapper">
  {#if $$slots.labelLeft}
    <label
      class="duk-button-label"
      class:duk-button-label--small="{size === sizes.BUTTON.SMALL}"
      class:duk-button-label--large="{size === sizes.BUTTON.LARGE}"
      class:duk-button-label--hero="{context === contexts.BUTTON.HERO}"
      for="{id}"
    >
      <slot name="labelLeft" />
    </label>
  {/if}
  {#if href && !disabled}
    <a href="{href}">
      <button
        class="{$$props.class || ''} duk-button"
        class:duk-button--brand="{variant === variants.BUTTON.BRAND}"
        class:duk-button--cta="{variant === variants.BUTTON.CTA}"
        class:duk-button--info="{variant === variants.BUTTON.INFO}"
        class:duk-button--success="{variant === variants.BUTTON.SUCCESS}"
        class:duk-button--warning="{variant === variants.BUTTON.WARNING}"
        class:duk-button--danger="{variant === variants.BUTTON.DANGER}"
        class:duk-button--small="{size === sizes.BUTTON.SMALL}"
        class:duk-button--large="{size === sizes.BUTTON.LARGE}"
        class:duk-button--outline="{outline}"
        class:duk-button--circle="{circle}"
        class:duk-button--block="{block}"
        class:duk-accordion__action="{context === contexts.BUTTON.ACCORDION}"
        class:duk-button--hero="{context === contexts.BUTTON.HERO}"
        class:duk-button--mnemonic="{context === contexts.BUTTON.MNEMONIC}"
        class:duk-button--control="{context === contexts.BUTTON.CONTROL}"
        class:duk-button--heading="{context === contexts.BUTTON.HEADING}"
        class:duk-button--group="{context === contexts.BUTTON.GROUP}"
        disabled="{disabled || undefined}"
        id="{id}"
        on:click="{() => (value = !value)}"
        on:click
        on:mouseover
        on:focus
      >
        <slot />
      </button>
    </a>
  {:else}
    <button
      class="{$$props.class || ''} duk-button"
      class:duk-button--brand="{variant === variants.BUTTON.BRAND}"
      class:duk-button--cta="{variant === variants.BUTTON.CTA}"
      class:duk-button--info="{variant === variants.BUTTON.INFO}"
      class:duk-button--success="{variant === variants.BUTTON.SUCCESS}"
      class:duk-button--warning="{variant === variants.BUTTON.WARNING}"
      class:duk-button--danger="{variant === variants.BUTTON.DANGER}"
      class:duk-button--small="{size === sizes.BUTTON.SMALL}"
      class:duk-button--large="{size === sizes.BUTTON.LARGE}"
      class:duk-button--outline="{outline}"
      class:duk-button--circle="{circle}"
      class:duk-button--block="{block}"
      class:duk-accordion__action="{context === contexts.BUTTON.ACCORDION}"
      class:duk-button--hero="{context === contexts.BUTTON.HERO}"
      class:duk-button--mnemonic="{context === contexts.BUTTON.MNEMONIC}"
      class:duk-button--control="{context === contexts.BUTTON.CONTROL}"
      class:duk-button--heading="{context === contexts.BUTTON.HEADING}"
      class:duk-button--group="{context === contexts.BUTTON.GROUP}"
      disabled="{disabled || undefined}"
      id="{id}"
      on:click="{() => (value = !value)}"
      on:click
      on:mouseover
      on:focus
    >
      <slot />
    </button>
  {/if}
  {#if $$slots.labelRight}
    <label
      class="duk-button-label"
      class:duk-button-label--small="{size === sizes.BUTTON.SMALL}"
      class:duk-button-label--large="{size === sizes.BUTTON.LARGE}"
      class:duk-button-label--hero="{context === contexts.BUTTON.HERO}"
      for="{id}"
    >
      <slot name="labelRight" />
    </label>
  {/if}
</div>
