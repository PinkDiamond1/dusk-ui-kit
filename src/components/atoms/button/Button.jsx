/* @flow */
import * as React from 'react'
import BootstrapButton from 'react-bootstrap/Button'

import './Button.scss'

export const ButtonTypes = {
  BUTTON: 'button',
  RESET: 'reset',
  SUBMIT: 'submit',
}

export const ButtonVariants = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  DANGER: 'danger',
}

export const ButtonSizes = {
  SMALL: 'sm',
  LARGE: 'lg',
}

type Props = {
  className: string,
  type: 'button' | 'reset' | 'submit',
  variant?: 'primary' | 'secondary' | 'info' | 'success' | 'warning' | 'danger',
  size?: 'small' | 'large',
  onClick: Function,
  children: React.Node,
  disabled?: boolean,
}

const Button = (props: Props): React.Element<*> => {
  const { className, type, onClick, children, variant, size, disabled } = props
  const updatedClassName = `__duk-button ${className}`
  return (
    <BootstrapButton
      className={updatedClassName}
      type={type}
      onClick={onClick}
      variant={variant}
      size={size}
      disabled={disabled}
    >
      {children}
    </BootstrapButton>
  )
}

Button.defaultProps = {
  type: ButtonTypes.BUTTON,
  variant: ButtonVariants.PRIMARY,
  size: '',
  onClick: () => {},
  className: '',
  disabled: false,
}

export default Button
