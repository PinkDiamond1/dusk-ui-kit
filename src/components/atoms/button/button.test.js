import React from 'react'
import renderer from 'react-test-renderer'

import Button from './Button'

it('test to see if the Button renders correctly', () => {
  const tree = renderer.create(<Button>Smoke Test</Button>).toJSON()
  expect(tree).toMatchSnapshot()
})
