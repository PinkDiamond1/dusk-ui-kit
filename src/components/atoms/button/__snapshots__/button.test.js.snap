// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Button renders correctly 1`] = `
<button
  className="__duk-button  btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Smoke Test
</button>
`;
